sdk/state/ingest.go

```diff
@@ -277,7 +277,7 @@ func (c *Channel) ingestFormationTx(tx *txnbuild.Transaction, resultXDR string,
 
 			switch updated.Data.Type {
 			case xdr.LedgerEntryTypeTrustline:
-				if updated.Data.TrustLine.Asset.StringCanonical() != c.openAgreement.Envelope.Details.Asset.StringCanonical() {
+				if updated.Data.TrustLine.Asset.ToAsset().StringCanonical() != c.openAgreement.Envelope.Details.Asset.StringCanonical() {
```

Comment:
💥 `ToAsset()` is unfortunately not safe to call on all asset types at the moment. If the trustline is a liquidity pool, it will panic.

See https://github.com/stellar/go/blob/5742327250c1468a68700a41c48ea16cb74dd8e4/xdr/asset.go#L257-L269.

I think we should add an `EqualsTrustLineAsset(ta xdr.TrustLineAsset)` method to the `state.Asset` type that will do a safer comparison. Wdyt?